python/ray/train/torch.py

```diff
@@ -289,52 +231,18 @@ def prepare_data_loader(
         move_to_device (bool): If set, automatically move the data
             returned by the data loader to the correct device.
     """
+    return get_session().accelerator.prepare_data_loader(
+        data_loader, add_dist_sampler=add_dist_sampler, move_to_device=move_to_device
+    )
 
-    # Only add Distributed Sampler if the following conditions hold:
-    # 1. More than one training worker is being used.
-    # 2. A DistributedSampler has not already been added by the user.
-    # 3. The dataset is not an IterableDataset. Samplers do not worker with
-    # IterableDatasets.
-    if (
-        train.world_size() > 1
-        and not isinstance(data_loader.sampler, DistributedSampler)
-        and not (
-            hasattr(data_loader, "dataset")
-            and isinstance(data_loader.dataset, IterableDataset)
-        )
-        and add_dist_sampler
-    ):
-
-        def with_sampler(loader):
-            # Automatically set the DistributedSampler
-
-            # If using a sampler, the shuffle attribute in the
-            # DataLoader must be set to False.
-            # Instead the shuffling is determined by the shuffle attribute
-            # in the DistributedSampler.
-            # We identify if shuffling is enabled in the passed in
-            # DataLoader by seeing if the sampler for the DataLoader is a
-            # SequentialSampler.
-            shuffle = not isinstance(loader.sampler, SequentialSampler)
-
-            data_loader_args = {
-                "dataset": loader.dataset,
-                "batch_size": loader.batch_size,
-                "shuffle": False,
-                "num_workers": loader.num_workers,
-                "collate_fn": loader.collate_fn,
-                "pin_memory": loader.pin_memory,
-                "drop_last": loader.drop_last,
-                "timeout": loader.timeout,
-                "worker_init_fn": loader.worker_init_fn,
-                "sampler": DistributedSampler(loader.dataset, shuffle=shuffle),
-            }
-            return DataLoader(**data_loader_args)
-
-        data_loader = with_sampler(data_loader)
-
-    if move_to_device:
-        device = get_device()
-        data_loader = _WrappedDataLoader(data_loader, device)
-
-    return data_loader
+
+@PublicAPI(stability="beta")
+def get_device() -> torch.device:
+    """Gets the correct torch device to use for training."""
+    return get_session().accelerator.get_device()
```

Comment:
Would it be clearer to extract `get_session().accelerator` into a `_get_or_create_accelerator` function?

Reply:
It would be more explicit, albeit more verbose. According to Zen of Python, your suggestion is more Pythonic

Reply:
I'm going to keep the implementation as is for now because `get_or_create_accelerator` is more verbose, and the return type of `get_or_create_accelerator` is potentially misleading. 

`get_or_create_accelerator` suggests that if there is no `Accelerator`, then the `Session` will create one. However, not all backends implement an `Accelerator`. For example, there is no `TensorflowAccelerator`. In these cases, `Session` will not create an `Accelerator` and `get_or_create_accelerator` will return `None`.

If we add `Session.get_or_create_accelerator`, we'd also need to add `Session.set_accelerator`. `Session.get_or_create_accelerator` and `Session.set_accelerator` is more verbose than `Session.attribute`.

I agree that constructing the `Accelerator` in the `Accelerator` getter is potentially confusing, but given that we'll likely refactor this implementation soon, I think it should be okay.